src/core/plugins/db-synchronizer.ts

```diff
@@ -0,0 +1,131 @@
+import {gatewayMethod, remoteApp, remoteMethod} from "./base/app-decorators.js";
+import PQueue from "p-queue";
+import CallablePlugin from "./base/callable-plugin.js";
+import AppManager from "./app-manager.js";
+import NodeManagerPlugin from "./node-manager.js";
+import * as NetworkIpc from "../../network/ipc.js";
+import {AppContext, MuonNodeInfo} from "../../common/types";
+import TssPlugin from "./tss-plugin.js";
+import {timeout} from "../../utils/helpers.js";
+import {logger} from '@libp2p/logger'
+
+const CONCURRENT_TSS_RECOVERY = 5;
+const log = logger("muon:core:plugins:synchronizer")
+
+const RemoteMethods = {
+  GetAllContexts: "get-all-ctx"
+}
+
+@remoteApp
+export default class DbSynchronizer extends CallablePlugin {
+  APP_NAME='synchronizer'
+  private readonly recoveryQueue: PQueue;
+
+  constructor(muon, configs) {
+    super(muon, configs)
+
+    this.recoveryQueue = new PQueue({
+      concurrency: CONCURRENT_TSS_RECOVERY,
+    });
+  }
+
+  async onStart(): Promise<void> {
+    await super.onStart();
+    log('onStart done.')
+
+    this.startMonitoring().catch(e => {});
+  }
+
+  private get nodeManager(): NodeManagerPlugin {
+    return this.muon.getPlugin('node-manager');
+  }
+
+  private get appManager(): AppManager {
+    return this.muon.getPlugin('app-manager');
+  }
+
+  private get tssPlugin(): TssPlugin {
+    return this.muon.getPlugin('tss-plugin')
+  }
+
+  private async startMonitoring() {
+    const {monitor: {startDelay, interval}} = this.muon.configs.net.synchronizer;
+    log(`monitor start %o`, {startDelay, interval})
+
+    await timeout(startDelay);
```

Comment:
Need to randomize this. Otherwise all nodes send request to deployers at the same time.
Sending requests after boot/restart also should be randomized.